plutus-tx-plugin/src/PlutusTx/Compiler/Types.hs

```diff
@@ -43,6 +45,12 @@ data CompileContext uni fun = CompileContext {
     ccBlackholed      :: Set.Set GHC.Name
     }
 
+-- | Profiling options. @All@ profiles everything. @None@ is the default.
+data ProfileOpts =
+    All -- set this with -fplugin-opt PlutusTx.Plugin:profile-all
+    | None
+    deriving (Eq)
```

Comment:
I guess it supposed to have more options later? Otherwise newtype over `Bool` might fit better.

Reply:
Yes, there may be more options later.